### PR TITLE
Error type for `WaitIdle`

### PR DESCRIPTION
**Pull request by GabrielMajeri**
**Connections**
Part of #638 

**Description**
Adds an error type for the `wait_idle` function.

**Testing**
Checked with core.

### CODE REVIEW

File: wgpu-core/src/device/life.rs [hunk 1]
```diff
@@ -259,25 +268,30 @@ impl<B: hal::Backend> LifetimeTracker<B> {
...
+            }
         }
+        Ok(())
     }
 
     /// Returns the last submission index that is done.
     pub fn triage_submissions(&mut self, device: &B::Device, force_wait: bool) -> SubmissionIndex {
         if force_wait {
-            self.wait_idle(device);
+            self.wait_idle(device).unwrap();
```
Comment (GabrielMajeri) [inline wgpu-core/src/device/life.rs:294]: I've looked into pulling this error out and returning it, but it's quite deep in the API. I don't know if it's desirable to do so.

**Comment by GabrielMajeri** [comment]
Not sure why the checks failed, the tests work on my end and there's no error log

File: wgpu-core/src/device/life.rs (hunk 1 above)
Comment (kvark) [reply wgpu-core/src/device/life.rs:294]: I think we'd need that, actually. Looking at the code that calls it
- `maintain` should return `Result`
- `prepare_to_die` should just log an error and early return, it doesn't need to unwrap or return a result
- `device_wait_for_buffer` should return `Result`

Comment (GabrielMajeri) [reply wgpu-core/src/device/life.rs]: Alright, I've made the required changes to get this error returned upwards.

**Pull request merged**